### PR TITLE
Lock store API docs

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- lock store and ticket lock documentation

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: docs/api/lock-stores.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+.. contents::
+
+InMemoryLockStore (default)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:Description:
+    ``InMemoryLockStore`` is the default lock store. It maintains conversation locks
+    within a single process.
+
+    .. note:: This lock store should not be used when multiple Rasa servers are run
```
Comment (tmbo) [inline docs/api/lock-stores.rst]: is the syntax correct here? might need a newline (could also be that the github syntax highlighting is off)

File: docs/api/lock-stores.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+RedisLockStore
+~~~~~~~~~~~~~~
+
+:Description:
+    ``RedisLockStore`` maintains conversation locks using Redis as a persistence layer.
+    This is the recommended lock store for running a replicated set of Rasa servers.
+
+:Configuration:
+    To set up Rasa Core with Redis the following steps are required:
```
Comment (tmbo) [inline docs/api/lock-stores.rst]:
```suggestion
    To set up Rasa with Redis the following steps are required:
```

File: docs/api/lock-stores.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    1. Start your Redis instance
+    2. Add required configuration to your `endpoints.yml`
```
Comment (tmbo) [inline docs/api/lock-stores.rst]:
```suggestion
    2. Add required configuration to your ``endpoints.yml``
```

File: docs/api/lock-stores.rst (hunk 1 above)
Comment (ricwo) [reply docs/api/lock-stores.rst]: 👍

**Pull request merged**